tests/framework/components/test_component.py

```diff
@@ -190,6 +204,26 @@ def test_component_that_creates_and_requires_columns_population_view():
     assert set(component.population_view.columns) == set(expected_columns)
 
 
+def test_component_with_initialization_requirements():
+    component = ColumnCreatorAndRequirer()
+    simulation = InteractiveContext(components=[ColumnCreator(), component])
+
+    # Assert required resources have been recorded by the ResourceManager
+    dependencies = [
+        r.dependencies
+        # get all resources in the dependency graph
+        for r in simulation._resource.sorted_nodes
+        # if the producer is an instance method
+        if hasattr(r.producer, "__self__")
```

Comment:
Can't you leave this line out and get the same thing with checking it's a method of ColumnCreatorAndRequirer alone?

Reply:
Are there resources that aren't instantiated objects?

Reply:
@hussain-jafari there are, but I don't care about them for this test; I only care about the resource created by `ColumnCreaterAndRequirer`.